### PR TITLE
Integrity check

### PR DESCRIPTION
**Pull request by VithikaS**
### Checklist

- [ ] I have read and understand the [contributing guidelines](../CONTRIBUTING.md#pull-requests)
- [ ] This PR fixes a defect, and I have provided tests to verify that the fix is effective
- [ ] This PR implements an enhancement, and I have provided tests to verify that it works as intended
- [ ] This PR introduces changes to the database model, and I have added corresponding [update logic](https://github.com/DependencyTrack/dependency-track/tree/master/src/main/java/org/dependencytrack/upgrade)
- [ ] This PR introduces new or alters existing behavior, and I have updated the [documentation](https://github.com/DependencyTrack/dependency-track/tree/master/docs/_docs) accordingly

### CODE REVIEW

File: src/main/java/org/dependencytrack/event/kafka/componentmeta/IntegrityCheck.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            LOGGER.debug("Integrity check is disabled");
+            return;
+        }
+        if (StringUtils.isEmpty(result.getComponent().getUuid())) {
+            LOGGER.info("Result received on topic does not have component uuid, integrity check cannot be performed");
+            return;
+        }
+        //if integritymetacomponent is  null, try to get it from db
+        //it could be that integrity metadata is already in db
+        IntegrityMetaComponent metadata = integrityMetaComponent == null ? qm.getIntegrityMetaComponent(result.getComponent().getPurl().toString()) : integrityMetaComponent;
```
Comment (sahibamittal) [inline src/main/java/org/dependencytrack/event/kafka/componentmeta/IntegrityCheck.java:37]: we can move metadata query after we check existence of component in db just below this.

File: src/main/java/org/dependencytrack/event/kafka/componentmeta/IntegrityCheck.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            integrityAnalysis = new IntegrityAnalysis();
+            integrityAnalysis.setComponent(component);
+        }
+        integrityAnalysis.setIntegrityCheckStatus(calculateIntegrityCheckStatus(md5Status, sha1Status, sha256Status, sha512Status));
+        integrityAnalysis.setMd5HashMatchStatus(md5Status);
+        integrityAnalysis.setSha1HashMatchStatus(sha1Status);
+        integrityAnalysis.setSha256HashMatchStatus(sha256Status);
+        integrityAnalysis.setSha512HashMatchStatus(sha512Status);
+        integrityAnalysis.setUpdatedAt(new Date());
+        qm.persist(integrityAnalysis);
```
Comment (sahibamittal) [inline src/main/java/org/dependencytrack/event/kafka/componentmeta/IntegrityCheck.java:60]: maybe move this data persisting inside an update method in a query manager class by sending whole updated object in parameter?

File: src/main/java/org/dependencytrack/event/kafka/componentmeta/IntegrityCheck.java (hunk 1 above)
Comment (VithikaS) [reply src/main/java/org/dependencytrack/event/kafka/componentmeta/IntegrityCheck.java:37]: Good spot, thanks

File: src/main/java/org/dependencytrack/event/kafka/componentmeta/IntegrityCheck.java (hunk 2 above)
Comment (VithikaS) [reply src/main/java/org/dependencytrack/event/kafka/componentmeta/IntegrityCheck.java:60]: Keeping it as is like we discussed

**Review by sahibamittal — APPROVED** [review]

**Review by mehab — APPROVED** [review]

**Pull request merged**